sway-error/src/error.rs

```diff
@@ -670,6 +670,11 @@ pub enum CompileError {
     RefMutableNotAllowedInMain { param_name: Ident },
     #[error("returning a `raw_ptr` from `main()` is not allowed")]
     PointerReturnNotAllowedInMain { span: Span },
+    #[error(
+        "Register \"{name}\" is initialized but later assigned which is not allowed. \
+            Consider uninitializing \"{name}\"."
```

Comment:
```suggestion
        "Register \"{name}\" is initialized and later reassigned which is not allowed. \
            Consider assigning to a different register inside the ASM block."
```
Rather than advising to not initialise, it might be more indicative to advice to use a different destination register inside the block.

Reply:
Yeah that makes sense. Thanks!